autotest.pm

```diff
@@ -177,7 +177,9 @@ sub loadtest ($script, %args) {
 
     my $nr = '';
     while (exists $tests{$fullname . $nr}) {
-        $test->{name} = join "#", $name, ++$nr;
+        my $new_name = join "#", $name, ++$nr;
+        $test->{name} = $new_name;
+        $test->{fullname} = "$category-$new_name";
```

Comment:
I guess the `$fullname` variable should also be updated, as it is used 3 lines below

Reply:
`$tests{$fullname . $nr} = $test;`

Indeed. It is missing the '#' character.

Reply:
Actually, I don't think it's the case. The loop statement `while (exists $tests{$fullname . $nr}) {` already expects that pattern. It's not clear to me where it `$tests` come from so I'll keep it as is.

Reply:
I was also confused about this suggestion when reviewing this change earlier and decided to not write anything to avoid writing something wrong :-)

So take the following with a grain of salt: You don't need to set `$fullname` here because considering how it is used below and in the loop condition it shouldn't contain `$nr`. So the change can stay as-is now that you have adapted tests.